File: julia/problem_06.jl
# Copyright 2014 Mitchell Kember. Subject to the MIT License.
# Project Euler: Problem 6
# Sum square difference

module problem_06

function solve()
	sum, sqsum = 0, 0
	for n = 1:100
		sum += n
		sqsum += n^2
	end
	sum^2 - sqsum
end

end # module

File: c++/src/problem_06.cpp
// Copyright 2016 Mitchell Kember. Subject to the MIT License.
// Project Euler: Problem 6
// Sum square difference

namespace problem_06 {

long sum_square_diff(const long n) {
	long sum = 0;
	long sq_sum = 0;
	for (long i = 1; i <= n; ++i) {
		sum += i;
		sq_sum += i * i;
	}
	return sum * sum - sq_sum;
}

long solve() {
	return sum_square_diff(100);
}

} // namespace problem_06
